src/main/java/com/devonfw/demoquarkus/logic/UcManageAnimalImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.devonfw.demoquarkus.logic;
+
+import com.devonfw.demoquarkus.domain.model.AnimalEntity;
+import com.devonfw.demoquarkus.domain.repo.AnimalRepository;
+import com.devonfw.demoquarkus.service.v1.mapper.AnimalMapper;
+import com.devonfw.demoquarkus.service.v1.model.AnimalDto;
+import com.devonfw.demoquarkus.service.v1.model.NewAnimalDto;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.transaction.Transactional;
+
+@Named
+@Transactional
+public class UcManageAnimalImpl implements UcManageAnimal {
+    @Inject
+    AnimalRepository animalRepository;
+
+    @Inject
+    AnimalMapper mapper;
+
+    @Override
+    public AnimalDto saveAnimal(NewAnimalDto dto) {
+        AnimalEntity created = this.animalRepository.save(this.mapper.create(dto));
+        return this.mapper.map(created);
+    }
+
+    @Override
+    public AnimalDto deleteAnimal(String id) {
+        AnimalEntity animal = this.animalRepository.findById(Long.valueOf(id)).get();
```

Comment:
why do we load an entity just to delete it?
That is a pure waste of performance.
Just deleteById that is already supported by `JpaRepository`.

Please try to implement deletion idempotent. So if you delete twice and 2nd time it is already deleted, do nothing (no exception).